Build failure due to unknown linker option on MacOS

Following the [README.md steps](https://github.com/SevenChords/CipesAtHome#macos) for source build on MacOS terminates during the final link with the following error:
```
/usr/local/opt/llvm/bin/clang -o recipesAtHome start.o inventory.o recipes.o config.o FTPManagement.o cJSON.o node_print.o calculator.o logger.o serialization.o shutdown.o -I/usr/local/include -L/usr/local/lib -lcurl -lconfig -fopenmp -I . -Wall -O2 -Wall -Werror=format-overflow -Werror=format-truncation -Werror=format-extra-args -Werror=format -Werror=maybe-uninitialized -Werror=array-bounds -Werror=narrowing -Werror=unknown-pragmas -Werror=multichar -Werror=implicit-function-declaration -Werror=implicit-int -Werror=incompatible-pointer-types -Werror=discarded-qualifiers  -Wl,--gc-sections
ld: unknown option: --gc-sections
clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [recipesAtHome] Error 1
```

#### Cause:
The ld64 linker used by MacOS does not have the option '--gc-sections' and instead uses '-dead_strip' as the flag to remove unused sections.

#### Tested Workaround:
Replace '--gc-sections' with '-dead_strip' at `Makefile:7`

#### Reproduction Steps:
1. `brew install llvm libomp libconfig` on a MacOS system with Homebrew installed
2. Clone CipesAtHome and checkout current Master branch
3. `make`

#### Environment
Commit: 75171f4126242c8cb2d28dab7b26a321cc3a2048
OS: MacOS 12.6
Clang version: Homebrew clang version 15.0.2 targeting x86_64-apple-darwin21.6.0

--- a/Makefile
+++ b/Makefile
@@ -17,6 +17,7 @@ ifeq ($(UNAME), Darwin)
 	MACPREFIX:=$(shell brew --prefix)
 	CC:=$(MACPREFIX)/opt/llvm/bin/clang
 	CFLAGS:=-I$(MACPREFIX)/include -L$(MACPREFIX)/lib $(CFLAGS)
+	FINAL_TARGET_CFLAGS:=$(subst --gc-sections,-dead_strip,$(FINAL_TARGET_CFLAGS))
 endif
 
 default: $(TARGET)

